perun-core/src/main/java/cz/metacentrum/perun/core/impl/Utils.java

```diff
@@ -152,68 +153,163 @@ public static String join(Iterable<?> collection, String separator) {
+
+					// Add additional rich user extSource
+					additionalUserExtSources.add(richUserExtSource);
 				}
 			}
 		}
 		return additionalUserExtSources;
 	}
 
+	/**
+	 * Returns loa of addtional ues, if not stated, returns 0.
+	 * Used in extractAdditionalUserExtSources to get ues LoA.
+	 *
+	 * @param login login of subject
+	 * @param userExtSourceRaw array containing LoA
+	 * @return int LoA
+	 */
+	private static int parseAdditionalUESLoa(String login, String[] userExtSourceRaw) {
+		int additionalExtLoa = 0;
+		// Loa is not mandatory argument
+		if (userExtSourceRaw.length>3 && userExtSourceRaw[3] != null) {
+			try {
+				additionalExtLoa = Integer.parseInt(userExtSourceRaw[3]);
+			} catch (NumberFormatException e) {
+				throw new ParserException("Subject with login [" + login + "] has wrong LoA '" + userExtSourceRaw[3] + "'.", e, "LoA");
+			}
+		}
+		return additionalExtLoa;
+	}
+
+	/**
+	 * Returns additional user ext source either found already in Perun or new.
+	 * Used in extractAdditionalUserExtSources to get ues.
+	 *
+	 * @param sess perun session
+	 * @param additionalExtSourceType type of additional ues
+	 * @param additionalExtSourceName name of additional ues
+	 * @param additionalExtLoa LoA of additional ues
+	 * @param additionalExtLogin login of additional ues
+	 * @return UserExtSource additional ues
+	 */
+	private static UserExtSource parseAdditionalUserExtSource(PerunSession sess, String additionalExtSourceType, String additionalExtSourceName, int additionalExtLoa, String additionalExtLogin) {
+		ExtSource additionalExtSource;
+		try {
+			// Try to get extSource, with full extSource object (containg ID)
+			additionalExtSource = ((PerunBl) sess.getPerun()).getExtSourcesManagerBl().getExtSourceByName(sess, additionalExtSourceName);
+		} catch (ExtSourceNotExistsException e) {
+			try {
+				// Create new one if not exists
+				additionalExtSource = new ExtSource(additionalExtSourceName, additionalExtSourceType);
+				additionalExtSource = ((PerunBl) sess.getPerun()).getExtSourcesManagerBl().createExtSource(sess, additionalExtSource, null);
+			} catch (ExtSourceExistsException e1) {
+				throw new ConsistencyErrorException("Creating existing extSource: " + additionalExtSourceName);
+			}
+		}
+		return new UserExtSource(additionalExtSource, additionalExtLoa, additionalExtLogin);
+	}
+
+	/**
+	 * Parses attributes of additional ues, which are in format: uesAttributeName=value1,value2. These attributes are returned with their values.
+	 * Used in extractAdditionalUserExtSources to get ues attributes.
+	 *
+	 * @param sess perun session
+	 * @param login login of subject
+	 * @param additionalExtLoginAndAttribute array containing extLogin and uesAttributes
+	 * @return List<Attribute> attributes of additional ues
+	 */
+	private static List<Attribute> parseUESAttributes(PerunSession sess, String login, String[] additionalExtLoginAndAttribute) {
```

Comment:
You can throw the part with attributes here and then take care of parsing and everything else (same as in the method above).